Tests/DistributedActorsTests/ClusterSystemTests.swift

```diff
@@ -378,29 +380,80 @@ final class ClusterSystemTests: ClusterSystemXCTestCase {
             throw testKit.fail("Expected timeout to be 200 milliseconds but was \(timeoutError.timeout)")
         }
     }
+
+    func test_remoteCall_interceptor() async throws {
+        let local = await setUpNode("local") { settings in
+            settings.enabled = true
+        }
+        let remote = await setUpNode("remote") { settings in
+            settings.enabled = true
+        }
+        local.cluster.join(node: remote.cluster.uniqueNode)
+
+        let greeter = Greeter(actorSystem: local, greeting: "hello")
+        let localGreeter = try Greeter.resolve(id: greeter.id, using: remote)
+
+        let otherGreeter = Greeter(actorSystem: local, greeting: "HI!!!")
+        localGreeter.id.context.remoteCallInterceptor = GreeterRemoteCallInterceptor(system: remote, greeter: otherGreeter)
```

Comment:
We can't do this -- context must not be mutated from the outside the actor.

Reply:
(For context here's some new actor system APIs to enable this code without assigning to context from outside https://github.com/yim-lee/swift-distributed-actors/commit/aa73aa1f6a2cf643c293306aa28f987d725b619d )